### PR TITLE
Use email api

### PR DESCRIPTION
**Pull request by gbonnefille**
This pull-request initiate a first usage of new API endpoints.

Using the official API can bring many interesting feature.

First; sing the official API will reduce code in this role. And less code == better code.

Furthermore, using an official API remove the need to deploy the dedicated scripts. As such deploy can be annoying depending on the overall deployment (native, docker, docker-compose, kubernetes), this will offer the ability to reuse easier this role in much more context.

Cf. #86 and #172

### CODE REVIEW

**Comment by zeitounator** [comment]
Thanks for this PR and for your interest. I will review shortly.

Using the API wherever possible is on my todo list. Not only to reduce code in the role (which has already drastically decreased in terms of groovy scripts), but also because sonatype change their internal API quite often (at least on the last releases....) which breaks the existing scripts.

An obvious candidate is the setup_ldap script which is blindly using an unpublished internal API...

Meanwhile, there are some place places in which we might want to keep the scripts for performance reasons (e.g. creating repos from a list, where the api will need a call for each repo). Furthermore, it is not always easy to reach idempotency with rest API without being very verbose.

File: defaults/main.yml [hunk 1]
```diff
@@ -72,6 +72,7 @@ nexus_public_hostname: 'nexus.vm'
 nexus_public_scheme: https
 
 # How should the role access the API for provisionning
+nexus_rest_api_endpoint_base: "service/rest"
```
Comment (zeitounator) [inline defaults/main.yml]: This name is conflicting/confusing. There is a version based detection of the rest endpoint in `nexus_install.yml` which sets `nexus_rest_api_endpoint` used in `call_script.yml, `declare_script_each.yml` and `nexus_install.yml`.

Unfortunately, at this point, the var is pointing directly to the `script` endpoint (which is historically the only one used).

To clean-up the role's code, I'm considering dropping support for nexus version < 3.11 which would remove this detection (along wiht some other path's detection) and would let us refactor the existing files to use a base as above for all api interactions.

**Comment by gbonnefille** [comment]
I pushed an other example of conversion (LDAP) in #254. I hope that help to design the next step for this role.

**Comment by gbonnefille** [comment]
Any news on this topic?

**Pull request closed without merging**